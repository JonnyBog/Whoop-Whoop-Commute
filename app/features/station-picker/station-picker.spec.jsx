import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import StationPicker from 'features/station-picker/station-picker';

describe('Features', () => {
  describe('Station Picker component', () => {
    it('should export a function', () => {
      expect(StationPicker).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        data: {
          matches: [
            {
              label: 'test label',
              value: 'test value'
            }
          ]
        },
        id: 'test id',
        setFieldValue: jest.fn(),
        requestStationPickerData: jest.fn(),
        error: 'test error'
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createShallowRenderTree
     * @param {Object} updatedProps - defaults to props
     * @returns {JSX} shallow renderer component
     */
    function createShallowRenderTree (updatedProps = props) {
      const shallowRenderer = rendererShallow.createRenderer();

      return shallowRenderer.render(
        <StationPicker {...updatedProps} />
      );
    }

    /**
     * createShallowWrapper
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} shallow component
     */
    function createShallowWrapper (updatedProps = props) {
      return shallow(
        <StationPicker {...updatedProps} />
      );
    }

    it('should return content', () => {
      const tree = createShallowRenderTree();
      expect(tree).toMatchSnapshot();
    });

    describe('Events', () => {
      it('should call setFieldValue with correct arguments when handleChange is called', () => {
        const wrapper = createShallowWrapper();
        wrapper.instance().handleChange({
          value: 'test'
        });

        expect(wrapper.instance().props.setFieldValue).toHaveBeenCalledWith(
          props.id,
          'test'
        );
      });

      it('should call requestStationPickerData with correct arguments when handleInputChange is called', () => {
        const wrapper = createShallowWrapper();
        wrapper.instance().handleInputChange({
          value: 'test'
        });

        expect(wrapper.instance().props.requestStationPickerData).toHaveBeenCalledWith(
          {
            value: 'test'
          }
        );
      });
    });
  });
});
