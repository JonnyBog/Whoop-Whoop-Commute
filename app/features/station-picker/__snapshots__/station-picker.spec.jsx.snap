// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Station Picker component should return content 1`] = `
<React.Fragment>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    id="test id"
    instanceId="test id"
    noOptionsMessage={[Function]}
    onChange={[Function]}
    onInputChange={[Function]}
    options={
      Array [
        Object {
          "label": "test label",
          "value": "test value",
        },
      ]
    }
    placeholder="Work station"
  />
  <FormError>
    test error
  </FormError>
</React.Fragment>
`;
